app/helpers/shifts_helper.rb

```diff
@@ -24,4 +24,9 @@ def officer_select_options(officers)
     officers.collect { |officer| [officer.name, officer.id] }
   end
 
+  private
+
+  def officer_shifts(task_id)
+    @officer.shifts.where(task: task_id).map(&:date)
```

Comment:
Thanks to rails Enumeration magic you could go `@officer.shifts.recount_scrutiny.map(&:date)` So if instead of passing around a `0` or `1` ... I would rethink this method to accept a symbol (`:recount_scrutiny` or `: vote_collection`) and transform this into `@officer.shifts.send(task.to_s).map(&:date)`

Reply:
It's ok, as we talked the scopes aren't working correctly and too much refactoring for this small task. We'll clean it up later 👍